dapps/tezosDomains.tsx

```diff
@@ -38,46 +44,253 @@ export const CLAIM_REVERSE_RECORD = {
+    [UPDATE_REVERSE_RECORD.mainnet]: true,
     [BUY_ADDRESS.mainnet]: true,
-    KT1EVYBj3f1rZHNeUtq4ZvVxPTs77wuHwARU: true,
-    KT1P8n2qzJjwMPbHJfi4o8xu6Pe3gaU3u2A3: true,
-    KT1CaSP4dn8wasbMsfdtGiCPgYFW7bvnPRRT: true,
-    KT1CfuAbJQbAGYcjKfvEvbtNUx45LY5hfTVR: true,
-    KT1MeFfi4TzSCc8CF9j3qq5mecTPdc6YVUPp: true,
+    [RENEW.mainnet]: true,
+    [COMMIT_ADDRESS.mainnet]: true,
+    [BID.mainnet]: true,
+    [WITHDRAW.mainnet]: true,
+    [SETTLE.mainnet]: true,
+    [OFFER.mainnet]: true,
     KT1GY5qCWwmESfTv9dgjYyTYs2T5XGDSvRp1: true,
-    KT1R4KPQxpFHAkX8MKCFmdoiqTaNSSpnJXPL: true,
+    [TOKEN_CONTRACT.mainnet]: true,
     KT1Lu5om8u4ns2VWxcufgQRzjaLLhh3Qvf5B: true,
     KT1VxKQbYBVD8fSkqaewJGigL3tmcLWrsXcu: true,
   },
   ghostnet: {
-    KT1B3j3At2XMF5P8bVoPD2WeJbZ9eaPiu3pD: true,
-    KT1HpddfW7rX5aT2cTdsDaQZnH46bU7jQSTU: true,
-    KT1Ln4t64RdCG1bK8zkH6Xi4nNQVxz7qNgyj: true,
-    KT1H19ouy5QwDBchKXcUw1QRFs5ZYyx1ezEJ: true,
-    KT1HDUc2xtPHqWQcjE1WuinTTHajXQN3asdk: true,
+    [CHECK_ADDRESS.ghostnet]: true,
+    [SET_CHILD_RECORD.ghostnet]: true,
+    [UPDATE_RECORD.ghostnet]: true,
+    [CLAIM_REVERSE_RECORD.ghostnet]: true,
+    [UPDATE_REVERSE_RECORD.ghostnet]: true,
     [BUY_ADDRESS.ghostnet]: true,
-    KT1Bv32pdMYmBJeMa2HsyUQZiC6FNj1dX6VR: true,
-    KT1PEnPDgGKyHvaGzWj6VJJYwobToiW2frff: true,
-    KT1P3wdbusZK2sj16YXxRViezzWCPXpiE28P: true,
-    KT1C7EF4c1pnPW9qcfNRiTPj5tBFMQJtvUhq: true,
-    KT1DMNPg3b3fJQpjXULcXjucEXfwq3zGTKGo: true,
+    [RENEW.ghostnet]: true,
+    [COMMIT_ADDRESS.ghostnet]: true,
+    [BID.ghostnet]: true,
+    [WITHDRAW.ghostnet]: true,
+    [SETTLE.ghostnet]: true,
+    [OFFER.ghostnet]: true,
+    [TOKEN_CONTRACT.ghostnet]: true,
   },
 };
 
-export function tezosDomains(transactions: Array<transaction>): CustomView {
+const promiseCache: {
+  [key: string]: Promise<ReactNode>;
+} = {};
+
+function getDomainPromise(
+  tokenId: number,
+  address: string,
+  Tezos: TezosToolkit
+) {
+  const cacheKey = `${address}:${tokenId}`;
+
+  if (!!promiseCache[cacheKey]) return promiseCache[cacheKey];
+
+  promiseCache[cacheKey] = getTokenMetadata(address, tokenId, Tezos)
+    .then(metadata => (
+      <a
+        href={`https://${
+          metadata.name?.endsWith(".gho") ? "ghostnet" : "app"
+        }.tezos.domains/domain/${metadata.name}`}
+        title="Open domain infos"
+        className="underline"
+        target="_blank"
+        rel="noreferre"
+      >
+        {metadata.name}
+      </a>
+    ))
+    .catch(_ => "Failed to fetch domain");
+
+  return promiseCache[cacheKey];
+}
+
+function PromiseRendrer({ promise }: { promise: Promise<React.ReactNode> }) {
+  const [{ node }, setPromiseState] = useState<{
+    isLoading: boolean;
+    node: React.ReactNode;
+  }>({
+    isLoading: true,
+    node: null,
+  });
+
+  useEffect(() => {
+    promise.then(node => {
+      setPromiseState({ isLoading: false, node });
+    });
+  }, [promise]);
+
+  return <>{node ?? "..."}</>;
```

Comment:
How about ?
```suggestion
  return <>{node ?? "loading ..."}</>;
```
Is it easy to add retry? It seems easy to fail?

Reply:
Imo, it's not that trivial to add it there